src/algoliasearch.helper.js

```diff
@@ -605,7 +605,7 @@ AlgoliaSearchHelper.prototype._getHitsSearchParams = function() {
     additionalParams.numericFilters = numericFilters;
   }
 
-  return extend(this.state.getQueryParams(), additionalParams);
+  return merge(this.state.getQueryParams(), additionalParams);
```

Comment:
Looking at the documentation, it doesn't seem like it could fail but it seems like it is a different and thus could bring unexpected results.